### PR TITLE
Fix `SystemString.withContiguousStorageIfAvailable()`

### PR DESCRIPTION
**Pull request by glessard**
- The original implementation simply forwarded to the underlying `nullTerminatedStorage`, which contains one element – a null terminator – more than the actual `String`.
- This isn’t in keeping with the sperit of `withContiguousStorageIfAvailable()`, which should present exactly the elements of the collection.

Addresses https://github.com/apple/swift-system/issues/153 (rdar://124102547)

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please test

File: Sources/System/SystemString.swift [hunk 1]
```diff
@@ -142,31 +142,38 @@ extension SystemString: RangeReplaceableCollection {
...
   }
 }
 
 extension SystemString: Hashable, Codable {}
 
 extension SystemString {
-  // TODO: Below include null terminator, is this desired?
 
+  // withSystemChars includes the null terminator
   internal func withSystemChars<T>(
```
Comment (lorentey) [inline Sources/System/SystemString.swift:173]: We should probably call this out in the name (`withNullTerminatedSystemChars` or something)

**Review by lorentey — APPROVED** [review]

File: Sources/System/SystemString.swift (hunk 1 above)
Comment (glessard) [reply Sources/System/SystemString.swift:173]: I'll include that change along with a consistency change for `withCodeUnits`.

Comment (glessard) [reply Sources/System/SystemString.swift:173]: Suggested name change included in https://github.com/apple/swift-system/pull/155

**Pull request merged**